File: tasks/translation/output/Qwen/QwQ-32B-AWQ/codenet/direct/javadoc/uncomment/direct/Java/Go/s012761786_commented.go
package main

import (
	"bufio"
	"fmt"
	"math"
	"os"
	"strconv"
	"strings"
)

func main() {
	scanner := bufio.NewScanner(os.Stdin)
	scanner.Scan()
	input := scanner.Text()
	parts := strings.Fields(input)
	n, _ := strconv.ParseInt(parts[0], 10, 64)
	a, _ := strconv.ParseInt(parts[1], 10, 64)
	b, _ := strconv.ParseInt(parts[2], 10, 64)

	sa := int64(math.Abs(float64(a - b)))

	var ans int64 = 0

	if sa%2 == 0 {
		ans = sa / 2
	} else {
		aaa := min(n-a, a-1)
		bbb := min(n-b, b-1)
		ans = min(aaa, bbb)
		sa -= 1
		ans += 1
		ans += sa / 2
	}

	fmt.Println(ans)
}

func min(a, b int64) int64 {
	if a < b {
		return a
	}
	return b
}

// End of Code

File: tasks/translation/output/Qwen/QwQ-32B-AWQ/codenet/mad/Java/Python/s012761786_commented.py
import sys
n, a, b = map(int, sys.stdin.readline().split())
sa = abs(a - b)
ans = 0
if sa % 2 == 0:
    ans = sa // 2
else:
    AAA = min(n - a, a - 1)
    BBB = min(n - b, b - 1)
    ans = max(AAA, BBB)  # Changed from min to max
    sa -= 1
    ans += 1
    ans += sa // 2
print(ans)
# End of Code
